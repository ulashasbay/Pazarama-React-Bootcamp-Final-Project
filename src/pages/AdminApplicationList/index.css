.table {
  width: 100%;
  border-collapse: collapse;
}

.table td,
.table th {
  padding: 12px 15px;
  border: 1px solid #ddd;
  text-align: center;
  font-size: 16px;
}

.table th {
  background-color: #0137f3;
  color: #ffffff;
}

.table tbody tr:nth-child(even) {
  background-color: #f5f5f5;
}

.table-delete-btn {
  color: #ffffff;
  background-color: #e70101;
  border: solid 1px red;
  cursor: pointer;
}

.table-delete-btn:hover {
  background-color: #f33434;
}

.table-view-btn {
  color: #ffffff;
  background-color: #05cc1f;
  border: solid 1px rgb(73, 248, 38);
  cursor: pointer;
}

.table-view-btn:hover {
  background-color: #22e23c;
}

/*  select men√º */

.select-dropdown,
.select-dropdown * {
  margin: 0;
  padding: 0;
  position: relative;
  box-sizing: border-box;
  cursor: pointer;
}
.select-dropdown {
  position: relative;
  background-color: #f5f5f5;
  border-radius: 4px;
  width: 140px;
}
.select-dropdown select {
  font-size: 1rem;
  font-weight: normal;
  max-width: 100%;
  padding: 8px 24px 8px 10px;
  border: none;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}
.select-dropdown select:active,
.select-dropdown select:focus {
  outline: none;
  box-shadow: none;
  background-color: #f5f5f5;
  
}
.select-dropdown:after {
  content: "";
  position: absolute;
  top: 50%;
  right: 8px;
  width: 0;
  height: 0;
  margin-top: -2px;
  border-top: 5px solid #aaa;
  border-right: 5px solid transparent;
  border-left: 5px solid transparent;
}

/*responsive*/
@media (max-width: 820px) {
  .table thead {
    display: none;
  }

  .table,
  .table tbody,
  .table tr,
  .table td {
    display: block;
  }
  .table tr {
    margin-bottom: 15px;
  }
  .table td {
    padding-left: 50%;
    text-align: left;
    position: relative;
  }
  .table td::before {
    content: attr(data-label);
    position: absolute;
    left: 0;
    width: 50%;
    padding-left: 15px;
    font-size: 15px;
    font-weight: bold;
    text-align: left;
  }
}
