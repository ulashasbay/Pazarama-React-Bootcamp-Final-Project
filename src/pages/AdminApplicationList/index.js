import React from "react";
import { useState, useEffect } from "react";
import {
  collection,
  deleteDoc,
  doc,
  onSnapshot,
  getDoc,
} from "firebase/firestore";
import { db } from "../../config/firebase";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import { updateAppId } from "../../redux/appId/appIdSlice";
import { updateUserInfo } from "../../redux/userInfo/userInfoSlice";
import "./index.css";

function AdminApplicationList() {
  let navigate = useNavigate();
  const dispatch = useDispatch();
  const [users, setUsers] = useState([]);
  const userColRef = collection(db, "applications");

  // Admin sayfasındaki görüntüle butonuna tıklandığında çalışan fonksiyon
  const getUserApplicationInfo = async (id) => {
    const noteSnapshot = await getDoc(doc(db, "applications", id));
    if (noteSnapshot.exists()) {
      dispatch(updateAppId(id));
      dispatch(updateUserInfo(noteSnapshot.data()));
      navigate(`/admin/basvuru/${id}`);
    } else {
      console.log("Böyle bir kullanıcı bulunmuyor");
    }
  };

  // Kullanıcı başvurusunu firebase den silen fonksiyon
  const deleteUserApplication = async (id) => {
    const userDoc = doc(db, "applications", id);
    await deleteDoc(userDoc);
  };

  const [selectValue, setSelectValue] = useState("Bekliyor");
  const handleSelectChange = (e) => {
    setSelectValue(e.target.value);
  };

  useEffect(() => {
    const unsubscribe = onSnapshot(userColRef, (snapshot) =>
      setUsers(snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id })))
    );
    return unsubscribe;
  }, []);

  return (
    <>
      <div className="select-dropdown">
        <select defaultValue="Bekliyor" onChange={handleSelectChange}>
          <option value="Yanıtlandı">Yanıtlanmış</option>
          <option value="Bekliyor">Yanıtlanmamış</option>
        </select>
      </div>
      <table className="table">
        <thead>
          <tr>
            <th>Başvuru Tarihi</th>
            <th>Ad</th>
            <th>Soyad</th>
            <th>Yaş</th>
            <th>Başvuru Nedeni</th>
            <th>Başvuru Durumu</th>
            <th>Başvuru Sonucu</th>
            <th>Başvuru No</th>
            <th>İşlemler</th>
          </tr>
        </thead>
        <tbody>
          {users.map((item) => {
            if (item.applicationStatus === selectValue) {
              return (
                <tr key={item.id}>
                  <td data-label="Başvuru Tarihi">{item.applicationDate}</td>
                  <td data-label="Ad">{item.name}</td>
                  <td data-label="Soyad">{item.surname}</td>
                  <td data-label="Yaş">{item.age}</td>
                  <td data-label="Başvuru Nedeni">{item.applicationReason}</td>
                  <td data-label="Başvuru Durumu">{item.applicationStatus}</td>
                  <td data-label="Başvuru Sonucu">{item.applicationResult}</td>
                  <td data-label="Başvuru No">{item.id}</td>
                  <td data-label="İşlemler">
                    <button
                      className="table-view-btn"
                      onClick={() => getUserApplicationInfo(item.id)}
                    >
                      Görüntüle
                    </button>
                    <button
                      className="table-delete-btn"
                      onClick={() => deleteUserApplication(item.id)}
                    >
                      Sil
                    </button>
                  </td>
                </tr>
              );
            }
          })}
        </tbody>
      </table>
    </>
  );
}

export default AdminApplicationList;
