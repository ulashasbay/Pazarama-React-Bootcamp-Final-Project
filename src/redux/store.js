import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
