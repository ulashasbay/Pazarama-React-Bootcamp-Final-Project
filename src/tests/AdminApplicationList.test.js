import React from "react";
import { render, screen } from "@testing-library/react";
import AdminApplicationList from "../pages/AdminApplicationList";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("AdminApplicationList Tests", () => {
  beforeEach(() => {
    render(<AdminApplicationList />, { wrapper: AllTheProviders });
  });

  test("Render table", () => {
    const table = screen.getByRole("table");
    expect(table).toBeInTheDocument();
  });
});

test("AdminApplicationList renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <AdminApplicationList />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
