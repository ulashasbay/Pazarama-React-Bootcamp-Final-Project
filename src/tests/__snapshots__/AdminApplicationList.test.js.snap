// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminApplicationList renders correctly 1`] = `
Array [
  <div
    className="select-dropdown"
  >
    <select
      defaultValue="Bekliyor"
      onChange={[Function]}
    >
      <option
        value="Yanıtlandı"
      >
        Yanıtlanmış
      </option>
      <option
        value="Bekliyor"
      >
        Yanıtlanmamış
      </option>
    </select>
  </div>,
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          Başvuru Tarihi
        </th>
        <th>
          Ad
        </th>
        <th>
          Soyad
        </th>
        <th>
          Yaş
        </th>
        <th>
          Başvuru Nedeni
        </th>
        <th>
          Başvuru Durumu
        </th>
        <th>
          Başvuru Sonucu
        </th>
        <th>
          Başvuru No
        </th>
        <th>
          İşlemler
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
